### DISCUSSION TITLE
Network to Code plugin onboarding for Netbox

### DISCUSSION THREAD

**Discussion by rfc-1149** [netbox-community/netbox#5538] [Q&A] [answered]
Hi everyone. 
I have installed Netbox following the official documentation. Everything works fine. Now I wish to install the **Network to Code onboarding plugin**. I am going through the installation as described on their Github page. This is what I have done:

```
cd /opt/netbox
source venv/bin/activate
pip install ntc-netbox-plugin-onboarding
systemctl restart netbox netbox-rq
echo ntc-netbox-plugin-onboarding >> local_requirements.txt
```
In your configuration.py
`PLUGINS = ["netbox_onboarding"]`

Everything fine until this point where I issue this command: 

`python3 manage.py migrate`

Then I get these errors:
**ModuleNotFoundError: No module named 'netbox_onboarding'
django.core.exceptions.ImproperlyConfigured: Unable to import plugin netbox_onboarding: Module not found. Check that the plugin module has been installed within the correct Python environment.**

I am new to Python and virtual environments. When I issue the command **pip list** inside the virtual environment I cannot see the network to code package. While outside the virtual environment I can see the **ntc-netbox-plugin-onboarding 2.0.0** being there. 
Is that the reason why it fails loading the module inside the virtual Python environment? I did install the package inside the virtual environment but not able to see it there. 

This is my first post at Github also.

**Comment by candlerb** [comment, marked as answer]
> When I issue the command pip list inside the virtual environment I cannot see the network to code package. While outside the virtual environment I can see the ntc-netbox-plugin-onboarding 2.0.0 being there.

It appears that you've installed ntc-netbox-plugin-onboarding outside the venv rather than inside it.  So something must have gone wrong at this point:

```
cd /opt/netbox
source venv/bin/activate
pip install ntc-netbox-plugin-onboarding
pip list | grep ntc
```

Can you show what happens when you run those commands, including the responses and command prompts you see from each one?

**Reply by rfc-1149** [reply]
After running the **pip install ntc-netbox-pluging-onboarding** I got this error:
**ERROR: Could not install packages due to an EnvironmentError: [Errno 13] Permission denied: '/opt/netbox/venv/lib/python3.8/site-packages/invoke'
Consider using the `--user` option or check the permissions.**

What I did then was to exit from the virtual environment and changed to the root user. Then went back into the virtual environment and issued the command again. This time it worked great. The plugin got installed. Was this the right thing to do? 

I believe that I also tried running the command with **sudo** in front of the command previously without succeding inside the virtual environment.  

Thank you so much for pushing me back into the right place again @candlerb

**Reply by candlerb** [reply]
> What I did then was to exit from the virtual environment and changed to the root user. Then went back into the virtual environment and issued the command again. This time it worked great. The plugin got installed. Was this the right thing to do?

Yes!

> I believe that I also tried running the command with **sudo** in front of the command previously without succeding inside the virtual environment.

"sudo" by default cleans the environment (i.e. environment variables).  You'd need `sudo -E pip install ...` to do it that way.

Glad you've got it to work anyway!
